src/lambda/expression.rs

```diff
@@ -75,13 +76,28 @@ impl Expression {
           match operation {
             Operation::Endpoint(req_template) => {
               let req = req_template.to_request(ctx)?;
-              let url = req.uri().clone();
+              let _url = req.uri().clone();
               let is_get = req.method() == reqwest::Method::GET;
+              let is_list = ctx
+                .context
+                .and_then(|c| c.ctx.path_node)
+                .and_then(|p| p.parent)
+                .map(|p| matches!(p.segment, QueryPathSegment::Index(_)))
+                .unwrap_or(false);
+
               // Attempt to short circuit GET request
-              if is_get {
-                if let Some(cached) = ctx.req_ctx.cache.get(&url) {
-                  return Ok(cached.body);
-                }
+              if is_get && is_list {
+                let headers = ctx.req_ctx.server.batch.clone().map(|s| s.headers).unwrap_or_default();
+                let endpoint_key = crate::http::EndpointKey::new(req, headers);
+                let resp = ctx
+                  .req_ctx
+                  .data_loader
+                  .as_ref()
+                  .load_one(endpoint_key)
+                  .await
+                  .map_err(|e| EvaluationError::IOException(e.to_string()))?
+                  .unwrap_or_default();
```

Comment:
MAJOR: Failures should be propagated, instead of responding with default value. 
This needs a unit test.